Crash when signing out and signing in again

This is new on `develop`, I tried on `release/9.0` and everything is fine.

1. Signin with a WPCom account via username + password -> 💥 .

Stack trace:
```
Process: org.wordpress.android.beta, PID: 3617
org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
    at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:505)
    at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:487)
    at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479)
    at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
    at java.lang.Thread.run(Thread.java:761)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'long org.wordpress.android.fluxc.model.SiteModel.getSelfHostedSiteId()' on a null object reference
    at org.wordpress.android.fluxc.network.xmlrpc.site.SiteXMLRPCClient.fetchProfile(SiteXMLRPCClient.java:44)
    at org.wordpress.android.fluxc.store.SiteStore.fetchProfileXmlRpc(SiteStore.java:828)
    at org.wordpress.android.fluxc.store.SiteStore.onAction(SiteStore.java:729)
    at java.lang.reflect.Method.invoke(Native Method)
    at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
    at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479) 
    at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
    at java.lang.Thread.run(Thread.java:761) 
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginUsernamePasswordFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginUsernamePasswordFragment.java
@@ -339,7 +339,7 @@ public class LoginUsernamePasswordFragment extends LoginBaseFormFragment<LoginLi
         });
     }
 
-    private @Nullable SiteModel detectNewlyAddedSite() {
+    private @Nullable SiteModel detectNewlyAddedXMLRPCSite() {
         List<SiteModel> selfhostedSites = mSiteStore.getSitesAccessedViaXMLRPC();
         for (SiteModel site : selfhostedSites) {
             if (!mOldSitesIDs.contains(site.getId())) {
@@ -429,6 +429,23 @@ public class LoginUsernamePasswordFragment extends LoginBaseFormFragment<LoginLi
         mLoginListener.loggedInViaPassword(mOldSitesIDs);
     }
 
+    private void finishLogin() {
+        AnalyticsUtils.trackAnalyticsSignIn(mAccountStore, mSiteStore, mIsWpcom);
+
+        startPostLoginServices();
+
+        // mark as finished so any subsequent onSiteChanged (e.g. triggered by WPMainActivity) won't be intercepted
+        mLoginFinished = true;
+
+        if (mLoginListener != null) {
+            if (mIsWpcom) {
+                saveCredentialsInSmartLock(mLoginListener.getSmartLockHelper(), mRequestedUsername, mRequestedPassword);
+            }
+
+            mLoginListener.loggedInViaUsernamePassword(mOldSitesIDs);
+        }
+    }
+
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(SiteStore.OnSiteChanged event) {
@@ -470,8 +487,13 @@ public class LoginUsernamePasswordFragment extends LoginBaseFormFragment<LoginLi
             return;
         }
 
-        SiteModel newlyAddedSite = detectNewlyAddedSite();
-        mDispatcher.dispatch(SiteActionBuilder.newFetchProfileXmlRpcAction(newlyAddedSite));
+        SiteModel newlyAddedXMLRPCSite = detectNewlyAddedXMLRPCSite();
+        // newlyAddedSite will be null if the user sign in with wpcom credentials
+        if (newlyAddedXMLRPCSite != null && !newlyAddedXMLRPCSite.isUsingWpComRestApi()) {
+            mDispatcher.dispatch(SiteActionBuilder.newFetchProfileXmlRpcAction(newlyAddedXMLRPCSite));
+        } else {
+            finishLogin();
+        }
     }
 
     @SuppressWarnings("unused")
@@ -495,20 +517,6 @@ public class LoginUsernamePasswordFragment extends LoginBaseFormFragment<LoginLi
             // continue with success, even if the operation was cancelled since the user got logged in regardless. So, go on
             //  with finishing the login process
         }
-
-        AnalyticsUtils.trackAnalyticsSignIn(mAccountStore, mSiteStore, mIsWpcom);
-
-        startPostLoginServices();
-
-        // mark as finished so any subsequent onSiteChanged (e.g. triggered by WPMainActivity) won't be intercepted
-        mLoginFinished = true;
-
-        if (mLoginListener != null) {
-            if (mIsWpcom) {
-                saveCredentialsInSmartLock(mLoginListener.getSmartLockHelper(), mRequestedUsername, mRequestedPassword);
-            }
-
-            mLoginListener.loggedInViaUsernamePassword(mOldSitesIDs);
-        }
+        finishLogin();
     }
 }

